### PR TITLE
ftgl: update to 2.1.3-rc5.

### PR DESCRIPTION
**Pull request by chissg**
This updates to the last release (from 2008) with patches required to work with Spack and Root.

These changes were implemented by @gartung on https://github.com/FNALssi/spack and abstracted for upstream by @chissg.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ftgl/package.py [hunk 1]
```diff
@@ -16,10 +14,15 @@ class Ftgl(AutotoolsPackage):
     list_url = "https://sourceforge.net/projects/ftgl/files/FTGL%20Source/"
     list_depth = 1
 
-    version('2.1.2', 'f81c0a7128192ba11e036186f9a968f2')
+    version('2.1.3-rc5', sha256='5458d62122454869572d39f8aa85745fc05d5518001bcefa63bd6cbb8d26565b')
 
-    # There is an unnecessary qualifier around, which makes modern GCC sad
-    patch('remove-extra-qualifier.diff')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/ftgl/package.py]: @chissg it looks like this patch isn't being used anymore: should it be removed from the Spack package dir?

Also regarding the new patch: could you add some info on it (e.g. where it came from if you didn't write it, or why you wrote it if you did).

File: var/spack/repos/builtin/packages/ftgl/package.py [hunk 2]
```diff
@@ -16,10 +14,15 @@ class Ftgl(AutotoolsPackage):
     list_url = "https://sourceforge.net/projects/ftgl/files/FTGL%20Source/"
     list_depth = 1
 
-    version('2.1.2', 'f81c0a7128192ba11e036186f9a968f2')
+    version('2.1.3-rc5', sha256='5458d62122454869572d39f8aa85745fc05d5518001bcefa63bd6cbb8d26565b')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/ftgl/package.py]: Note (not a request): removing old versions is generally discouraged but ROOT is currently the only package using this.

File: var/spack/repos/builtin/packages/ftgl/package.py (hunk 1 above)
Comment (chissg) [reply var/spack/repos/builtin/packages/ftgl/package.py]: I have restored `2.1.2` as an acceptable version, which once again uses this patch. I have also added a `doc` variant to allow documentation to be optionally produced for `2.1.2`. Unfortunately, `2.1.3-rc5` appears to have a broken `doc` step so the documentation generation is unconditionally disabled for this version.

File: var/spack/repos/builtin/packages/ftgl/package.py (hunk 2 above)
Comment (chissg) [reply var/spack/repos/builtin/packages/ftgl/package.py]: See https://github.com/spack/spack/pull/11214#discussion_r277428195

**Pull request merged**